devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -2253,91 +2367,114 @@ bool HumanStateProvider::impl::solveDynamicalInverseKinematics()
+                        linkName, wearableTargetEntry.second->getCalibratedRotation())) {
+                    yError() << LogPrefix << "Failed to update orientation target for "
+                             << targetName;
+                    return false;
+                }
+                break;
+            }
+            case hde::KinematicTargetType::orientationAndVelocity: {
+                if (!dynamicalInverseKinematics.updateTargetOrientationAndVelocity(
+                        linkName,
+                        wearableTargetEntry.second->getCalibratedRotation(),
+                        wearableTargetEntry.second->getCalibratedAngularVelocity())) {
+                    yError() << LogPrefix << "Failed to update orientation and velocity target for "
+                             << targetName;
+                    return false;
+                }
+                break;
+            }
+            case hde::KinematicTargetType::gravity: {
+                if (!dynamicalInverseKinematics.updateTargetOrientation(
+                        linkName, wearableTargetEntry.second->getCalibratedRotation())) {
+                    yError() << LogPrefix << "Failed to update gravity target for " << targetName;
+                    return false;
+                }
+                break;
+            }
+            case hde::KinematicTargetType::floorContact: {
+                if (!dynamicalInverseKinematics.updateTargetPositionAndVelocity(
+                        linkName,
+                        wearableTargetEntry.second->getCalibratedPosition(),
+                        wearableTargetEntry.second->getCalibratedLinearVelocity())
+                    || !dynamicalInverseKinematics.updatePositionTargetAxis(
+                        linkName,
+                        {wearableTargetEntry.second->contactActive,
+                         wearableTargetEntry.second->contactActive,
+                         wearableTargetEntry.second->contactActive})) {
+
+                    yError() << LogPrefix << "Failed to update position and velocity target for "
+                             << targetName;
+                    return false;
+                }
+                break;
+            }
+            default: {
+                yError() << LogPrefix << "Invalid target type for " << targetName;
                 return false;
-                                                       }
-            break;
-        }
-        default: {
-            yError() << LogPrefix << "Invalid target type for " << targetName;
-            return false;}
+            }
         }
     }
 
-    if (!dynamicalInverseKinematics.solve(dt))
-        return false;
-    
-    dynamicalInverseKinematics.getConfigurationSolution(baseTransformSolution, jointConfigurationSolution);
+    {
+        std::lock_guard<std::mutex> lock(mutexFlagIntegrator);
+        if (!dynamicalInverseKinematics.solve(dt, resetIntegrator))
+            return false;
```

Comment:
```suggestion
        if (!dynamicalInverseKinematics.solve(dt, resetIntegrator))
        {
            return false;
        } 
        else 
        {
            // Reset was successufl, we can set resetIntegrator back to false
            resetIntegrator = false;
        }
```

I find this more readable, what do you think @dariosortino @davidegorbani ? Note that in this case all the other arguments can be changed from `bool &` to `const bool`, I did not do all the modifications.